src/workers/forum_automod.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime, timezone, timedelta
+import discord
+import logging
+import asyncio
+from settings import get_settings
+from util import log_info_and_embed, send_dm
+from .helper import create_automod_embed
+from discord.ext import tasks
+from discord.utils import snowflake_time
+
+settings = get_settings()
+logger = logging.getLogger(__name__)
+
+
+@tasks.loop(hours=24)
+async def autodelete_threads(self):
+    guild = self.get_guild(settings.guild_id)
+    if guild is None:
+        return
+
+    for channel_id, duration in settings.automod_inactivity.items():
+        channel = guild.get_channel(channel_id)
+        if channel is None:
+            continue
+
+        for thread in channel.threads:
+            last_post = thread.last_message_id
+
+            if thread.flags.pinned:
+                continue
+
+            # check if starter message was deleted
+            starter_message = None
+            try:
+                starter_message = await thread.fetch_message(thread.id)
+                await asyncio.sleep(0.5)
+            except discord.NotFound:
+                pass
+            except Exception as e:
+                logger.error(e)
+
+            now = datetime.now(timezone.utc)
+            time_since = now - snowflake_time(last_post)
+            if starter_message is not None and time_since < timedelta(days=duration):
+                continue
+
+            # delete thread and try to send DM to user
+            try:
+                async with create_automod_embed(
+                    self,
+                    thread,
+                    now,
+                ) as automod_embed:
+                    await thread.delete()
+                    log_info_and_embed(
+                        automod_embed, logger, "Thread has been deleted successfully."
+                    )
+
+                    # TODO: uncomment DM portion and sleep when backlog is dealt with
+                    # if thread.owner is not None:
```

Comment:
i think we won't ever hit this case because there will always be an owner for a thread

Reply:
the `thread.owner` property returns `Optional[discordgo.Member]`. I believe it will return `None` when the person has left the server.

Reply:
uh hold on, discordgo shouldnt be an object in this case, are we looking at the right documentation?

Reply:
sorry, I specified discordgo.Member since that's how I'd refer tot he class in the code

Reply:
ok that sounds good to me. if we want, we could be more specific in the errors of DMs being turned off vs user is not in the discord server anymore

Reply:
oh right, I meant to ask this too. should I add a case for that or is letting the embed handle it good enough?
![image](https://github.com/user-attachments/assets/063163d0-f818-4d76-a11a-6148cf84a6ea)

Reply:
I'm just gonna handle it to make things consistent actually